### PR TITLE
Change Martha (v3) to use DRS (not DOS) endpoints for Gen3 projects (BDCat and AnVIL) [BT-158]

### PR DESCRIPTION
**Pull request by grsterin**

### CODE REVIEW

**Review by kshakir — APPROVED** [review]

**Review by mcovarr — COMMENTED** [review]
Multiple test failures though the Jenkins log output is not especially informative...

**Comment by grsterin** [comment]
> Multiple test failures though the Jenkins log output is not especially informative...

Yes, looks like those are integration tests' failures but I don't know why they happen. Also I tried restarting Jenkins builds but they just fail immediately now. I guess I need help from @kshakir to figure this out.

**Comment by kshakir** [comment]
You can also check with Batch Workflows. They worked on "fixing" the equivalent tests for the BT team over in the Cromwell repo. Where the BW "fix" equaled removing the tests. 😭 (I would link to the Jira... but BW crippled the tests without filing a ticket.) 

Tests that use credentials have been broken for weeks even in Cromwelll. The credentials need to be coordinated with the workbench team before you can proceed. For further context, this is the stuff the Workbench Sprint Reviews have been advertising for months-- fixes in permission models across Terra/Sam/TDR.

**TL;DR Permissions changed in Workbench. They need to be updated in CI for our various repos. Or the tests may just be deleted like before and you can beg forgiveness from Ambica et al.**

**Comment by kshakir** [comment]
Update: dunno who/what/where/why/when the other CI issues cleared up, but running CI tests locally I think the remaining issues at this point are due to DOS-to-DRS updates required over in [this corner](https://github.com/broadinstitute/mock-drs) of the workbench multi-repo.

We can chat more during "DRS HCA/TDR tests' weirdness" tomorrow.

In the meantime, jenkins retest

**Comment by kshakir** [comment]
Yep, [remaining failures](https://fc-jenkins.dsp-techops.broadinstitute.org/job/martha-fiab-test-runner/273/console) match with what I saw locally:

<details>
<summary>2 tests failed</summary>

  2 tests failed  martha › integration_v3 › integration_v3 succeeds for a protected url

    ℹ {
        response: {
          status: 403,
          text: 'Received error while resolving DRS URL. Forbidden',
        },
        status: 403,
      }

  Callback called with an error:

  AssertionError {
    actual: 403,
    code: 'ERR_ASSERTION',
    expected: 200,
    generatedMessage: false,
    operator: 'strictEqual',
    message: 'Incorrect status code',
  }



  martha › integration_v3 › integration_v3 succeeds for a public url

    ℹ {
        response: {
          status: 403,
          text: 'Received error while resolving DRS URL. Forbidden',
        },
        status: 403,
      }

  Callback called with an error:

  AssertionError {
    actual: 403,
    code: 'ERR_ASSERTION',
    expected: 200,
    generatedMessage: false,
    operator: 'strictEqual',
    message: 'Incorrect status code',
  }

  ─

  2 tests failed


</details>

**Comment by grsterin** [comment]
jenkins retest

**Review by mcovarr — APPROVED** [review]
with Jenkins' blessing
